docs/calculating-effective-entry-date.md

```diff
@@ -0,0 +1,163 @@
+
+# Calculating Effective Entry Dates Using moov-io/base Functions
+
+## Overview
+
+The moov-io/base package provides functionality for calculating valid Effective Entry Dates when creating ACH batch headers, ensuring compliance with NACHA rules and Federal Reserve processing schedules.
+
+## Creating Batch Headers
+
+The NewBatchHeader function creates a BatchHeader object that requires several key fields to be populated according to NACHA rules:
+
+```go
+bh := NewBatchHeader()
+bh.ServiceClassCode = "220"                  // 220 for credits only
+bh.StandardEntryClassCode = PPD              // PPD, CCD, WEB, etc.
+bh.CompanyName = "ACME Corporation"          // Your company name
+bh.CompanyIdentification = "121042882"       // Your company ID
+bh.CompanyEntryDescription = "PAYROLL"       // Transaction description
+bh.ODFIIdentification = "12104288"           // Your bank's routing number
+
+bh.EffectiveEntryDate = "240102"             // YYMMDD format
+```
+
+## Banking Day Functions
+
+The moov-io/base Time package provides these essential functions for determining valid banking days:
+
+* `AddBankingTime(hours, minutes, seconds int)`: Increments time by specified duration while respecting banking hours (9am-5pm ET)
+* `AddBankingDay(d int)`: Adds the specified number of valid banking days to a date
+* `IsBankingDay()`: Validates if a date is a banking day by checking both holidays and weekends
+* `IsBusinessDay()`: Determines if a date is a standard business day
+* `IsHoliday()`: Checks if a date falls on a Federal Reserve holiday
+* `IsWeekend()`: Verifies if a date falls on Saturday or Sunday
+
+## Transmission Windows
+
+The package handles Same-Day ACH transmission windows with time-based validations. Here's how to implement the cutoff time checks:
+
+```go
+now := base.Now()
+// Create cutoff times for each window
+firstCutoff := now.AddBankingTime(10, 30, 0)    // 10:30 AM ET
+secondCutoff := now.AddBankingTime(14, 45, 0)   // 2:45 PM ET
+thirdCutoff := now.AddBankingTime(16, 45, 0)    // 4:45 PM ET
+
+if now.Before(firstCutoff.Time) {
+   // In first window
+   bh.EffectiveEntryDate = now.Format("060102")
+} else if now.Before(secondCutoff.Time) {
+   // First window closed, in second window
+   bh.EffectiveEntryDate = now.Format("060102")
+} else if now.Before(thirdCutoff.Time) {
+   // Second window closed, in third window
+   bh.EffectiveEntryDate = now.Format("060102")
+} else {
+   // All windows closed for today
+   now = now.AddBankingDay(1)
+   bh.EffectiveEntryDate = now.Format("060102")
+}
+```
+
+## Implementation Example
+
+Here's a complete example showing how to calculate an Effective Entry Date based on various conditions:
+
+```go
+func calculateEffectiveEntryDate(amount int, isCredit bool) string {
+   now := base.Now()
+   
+   // On weekends and holidays, entries must be processed on future banking days
+   if !now.IsBankingDay() {
```

Comment:
Internally at Moov we made this recursive so it advances past weekends / holidays (especially Friday/Monday holidays). I'd probably invert the logic to check `if now.IsBankingDay() { ... }` as the current code doesn't seem correct.

Reply:
Rewrote this function to be shorter and simpler while checking if the current day is a banking day.